compiler/src/Language/Bond/Codegen/Cpp/Types_cpp.hs

```diff
@@ -20,16 +21,18 @@ types_cpp :: MappingContext -> String -> [Import] -> [Declaration] -> (String, T
 types_cpp cpp file _imports declarations = ("_types.cpp", [lt|
 #include "#{file}_reflection.h"
 #include <bond/core/exception.h>
-#include <unordered_map>
-
+#{ifThenElse (anyEnum declarations) unorderedMapInclude mempty}
 #{CPP.openNamespace cpp}
     #{doubleLineSepEnd 1 statics declarations}
 #{CPP.closeNamespace cpp}
 |])
   where
+    unorderedMapInclude = [lt|#include <unordered_map>
+|]
+
     -- definitions of Schema statics for non-generic structs
     statics s@Struct {..} =
-        if null declParams then CPP.schemaMetadata cpp s else mempty
+        ifThenElse (null declParams) (CPP.schemaMetadata cpp s) mempty
```

Comment:
I find it clearer with the `if ... then ... else` like before. We don't need a function call here, so the expression form is fine.

Reply:
Reverted.